pkg/metrics/collector/kernel_metrics.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	tcpTimeoutDesc = prometheus.NewDesc(
+		"tcp_timeout_rehash",
+		"Tcp timeout rehash, collected from nstat",
+		nil, nil,
+	)
+	tcpDuplicateDesc = prometheus.NewDesc(
+		"tcp_duplicate_rehash",
+		"Tcp duplicate rehash, collected from nstat",
+		nil, nil,
+	)
+	segmentsReceivedDesc = prometheus.NewDesc(
+		"segments_received",
+		"Segments received on the node, collected from netstat -s",
+		nil, nil,
+	)
+	segmentsSentDesc = prometheus.NewDesc(
+		"segments_sent",
+		"Segments sent from node, collected from netstat -s",
+		nil, nil,
+	)
+	segmentsRetransmittedDesc = prometheus.NewDesc(
+		"segments_retransmitted",
+		"Segments retransmitted on the node, collected from netstat -s",
+		nil, nil,
+	)
+)
+
+type kernelStatCollector struct {
+}
+
+func init() {
+	registerCollector("kernel_metrics", NewKernelStatCollector)
+}
+
+func NewKernelStatCollector() (Collector, error) {
+	return &kernelStatCollector{}, nil
+}
+
+type kernelStats struct {
+	tcpTimeoutRehash      uint64
+	tcpDuplicateRehash    uint64
+	segmentsReceived      uint64
+	segmentsSent          uint64
+	segmentsRetransmitted uint64
+}
+
+func parseNetstatVals(stats *kernelStats, output string) error {
```

Comment:
Consider returning the kernelStats struct as output.

I'm actually not sure why you have it here as a param, let me know if I am missing something.

Reply:
Main reason I had it as a param was because there are 2 functions which fill this struct. However I think it migh be better to declare two structs, one for nstat and one for netstat. Let me know what you think.